command/agent/command.go

```diff
@@ -590,10 +674,25 @@ func (c *Command) retryJoin(config *Config, errCh chan<- struct{}) {
 
 	attempt := 0
 	for {
-		n, err := c.agent.JoinLAN(config.RetryJoin)
-		if err == nil {
-			logger.Printf("[INFO] agent: Join completed. Synced with %d initial agents", n)
-			return
+		var servers []string
+		var err error
+		if ec2Enabled {
+			servers, err = config.discoverEc2Hosts(logger)
+			if err != nil {
+				logger.Printf("[ERROR] agent: Unable to query EC2 insances: %s", err)
+			}
+			logger.Printf("[INFO] agent: Discovered %d servers from EC2...", len(servers))
+		}
+
+		servers = append(config.RetryJoin, servers...)
```

Comment:
This will work fine, but I'd flip this to keep the servers so far in front, so `servers = append(servers, config.RetryJoin...)`.